src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -530,6 +545,27 @@ void SettingsLoader::_appendProfile(winrt::com_ptr<Profile>&& profile, ParsedSet
     }
 }
 
+// We offer a global option called "disabledProfileSources" (it's an array of strings).
+// If the option contains "Windows.Terminal.Wsl" for instance, we'll not generate any dynamic WSL profiles.
+// 
+// Unfortunately this poses a problem: What about existing user profiles?
+// This function ensures that existing profiles with such disabled sources are removed.
+void SettingsLoader::_removeIgnoredProfiles(ParsedSettings& settings)
+{
+    const auto begin = settings.profiles.begin();
+    const auto end = settings.profiles.end();
+    const auto it = std::remove_if(begin, end, [&](const auto& profile) {
+        const auto source = profile->Source();
+        const auto remove = !source.empty() && _ignoredNamespaces.count(source);
+        if (remove)
+        {
+            settings.profilesByGuid.erase(profile->Guid());
+        }
+        return remove;
+    });
+    settings.profiles.erase(it, end);

```

Comment:
do we need to adjust the userProfilesCount?

Reply:
`_userProfilesCount` is calculated after this (private) function is called.